samples/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgismaps/sample/editfeatureswithfeaturelinkedannotation/screens/EditFeaturesWithFeatureLinkedAnnotationScreen.kt

```diff
@@ -0,0 +1,162 @@
+                    .fillMaxSize()
+                    .padding(padding)
+            ) {
+                Box(modifier = Modifier.fillMaxSize()) {
+                    MapView(
+                        modifier = Modifier
+                            .fillMaxSize(),
+                        arcGISMap = mapViewModel.arcGISMap,
+                        mapViewProxy = mapViewModel.mapViewProxy,
+                        onSingleTapConfirmed = { tapEvent ->
+                            tapEvent.mapPoint?.let { mapPoint ->
+                                mapViewModel.onMapSingleTap(
+                                    screenCoordinate = tapEvent.screenCoordinate,
+                                    mapPoint = mapPoint
+                                )
+                            }
+                        }
+                    )
+                    Box(
+                        modifier = Modifier
+                            .align(Alignment.TopCenter)
+                            .padding(8.dp)
+                            .clip(RoundedCornerShape(10.dp))
+                            .background(Color.Black.copy(alpha = 0.8f))
+                            .padding(16.dp),
+                        contentAlignment = Alignment.Center
+                    ) {
+                        Text(
+                            text = currentInstruction.message,
+                            color = Color.White
+                        )
+                    }
+                }
+                if (mapViewModel.showEditAddressDialog) {
+                    EditAddressDialog(
+                        buildingNumber = mapViewModel.buildingNumber.toString(),
+                        streetName = mapViewModel.streetName,
+                        onBuildingNumberChange = { mapViewModel.onBuildingNumberChange(it.toInt()) },
+                        onStreetNameChange = { mapViewModel.onStreetNameChange(it) },
+                        onDismiss = { mapViewModel.onShowEditAddressDialogChange(false) },
+                        onConfirm = {
+                            mapViewModel.onEditAddressConfirmed(
+                                mapViewModel.selectedFeature,
+                                mapViewModel.buildingNumber,
+                                mapViewModel.streetName
+                            )
+                        }
+                    )
+                }
+                mapViewModel.messageDialogVM.apply {
+                    if (dialogStatus) {
+                        MessageDialog(
+                            title = messageTitle,
+                            description = messageDescription,
+                            onDismissRequest = ::dismissDialog
+                        )
+                    }
+                }
+            }
+        }
+    )
+}
+
+@Composable
+fun EditAddressDialog(
+    buildingNumber: String,
+    streetName: String,
+    onBuildingNumberChange: (String) -> Unit,
```

Comment:
Clearing the string here will cause the app to crash as an empty string cannot be cast to an int:

```suggestion
    onBuildingNumberChange: (Int) -> Unit,
```